sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java

```diff
@@ -582,29 +608,85 @@ public long getTotalBacklogBytes() {
     }
 
     @Override
-    public void close() throws IOException {
+    public void close() {
+      doClose();
+    }
+
+    @SuppressWarnings("FutureReturnValueIgnored")
+    private void doClose() {
+      if (active.get()) {
+        try {
+          closeAutoscaler();
+          closeConsumer();
+          ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
```

Comment:
Once https://github.com/apache/beam/pull/23234/files gets merged then you can use the `ScheduledExecutorService` from `ExecutorOptions` instead of creating one.

Reply:
Ok